indra_db/readonly_dumping/readonly_dumping.py

```diff
@@ -0,0 +1,2089 @@
+    logger.info(f"Building index on {table.full_name()}")
+    table.build_indices(local_ro_mngr)
+
+    logger.info(f"Deleting {split_pa_link_folder_fpath.absolute().as_posix()}")
+    shutil.rmtree(split_pa_link_folder_fpath.absolute().as_posix())
+    logger.info(f"Deleting {pa_hash_act_type_ag_count_cache.absolute().as_posix()}")
+    os.remove(pa_hash_act_type_ag_count_cache.absolute().as_posix())
+
+
+def ensure_source_meta_source_files(local_ro_mngr: ReadonlyDatabaseManager):
+    """Generate the source files for the SourceMeta table"""
+    if not table_has_content(local_ro_mngr, "name_meta"):
+        raise ValueError(
+            "name_meta must be filled before source_meta can be filled"
+        )
+    col_names = "ev_count, belief, num_srcs, " \
+                "src_json, only_src, has_rd, has_db, " \
+                "type_num, activity, is_active, agent_count"
+    all_sources = list(
+        {*SOURCE_GROUPS["reader"], *SOURCE_GROUPS["database"]}
+    )
+    all_sources_str = ", ".join(all_sources)
+
+    if source_meta_parquet.exists():
+        return "mk_hash, " + all_sources_str + ", " + col_names, all_sources
+    logger.info("Loading source counts")
+    # Load source_counts
+    ro = local_ro_mngr
+    source_counts = pickle.load(source_counts_fpath.open("rb"))
+
+    # Dump out from NameMeta
+    # mk_hash, type_num, activity, is_active, ev_count, belief, agent_count
+    # where is_complex_dup == False
+    # todo If it's too large, we can do it in chunks of 100k
+    # Typical count for namemeta 90_385_263
+    res = local_ro_mngr.select_all([ro.NameMeta.mk_hash,
+                                    ro.NameMeta.ev_count,
+                                    ro.NameMeta.belief,
+                                    ro.NameMeta.type_num,
+                                    ro.NameMeta.activity,
+                                    ro.NameMeta.is_active,
+                                    ro.NameMeta.agent_count],
+                                   ro.NameMeta.is_complex_dup == False)
+
+    # Loop the dump and create the following columns as well:
+    # source count, src_json (==source counts dict per mk_hash), only_src (if
+    # only one source), has_rd (if the source is in the reading sources),
+    # has_db (if the source is in the db sources)
+    mk_hash_set = set()
+    rows = []
+    logger.info("Generating source_meta parquet")
+    for (mk_hash, ev_count, belief, type_num,
+         activity, is_active, agent_count) in res:
+        if mk_hash in mk_hash_set:
+            continue
+        mk_hash_set.add(mk_hash)
+        # Get the source count
+        src_count_dict = source_counts.get(mk_hash)
+        if src_count_dict is None:
+            continue
+
+        num_srcs = len(src_count_dict)
+        has_rd = any(source in src_count_dict for source in SOURCE_GROUPS["reader"])
+        has_db = any(source in src_count_dict for source in SOURCE_GROUPS["database"])
```

Comment:
We have to ensure mapping from indra db name works with indra names and that the distinction between database/reader is correct here, since the source counts use the names found in DBInfo table, while the `SOURCE_GROUPS` contains names as they are in indra I'll take a look at this as well.

Reply:
Updates made in dc8df595fcb53b11b4bdc9cba79ea24469fe6bc2